### PR TITLE
Location property must have x-ms-mutability extension

### PR DESCRIPTION
**Pull request by dsgouda**
Addresses https://github.com/Azure/autorest/issues/1961

### CODE REVIEW

**Comment by dsgouda** [comment]
@azuresdkci Test this please

File: src/modeler/AutoRest.Swagger/Validation/LocationMustHaveXmsMutabilityExt.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> definitions, RuleContext context)
+        {
+            var serviceDefinition = (ServiceDefinition)context.Root;
+            var trackedResources = context.TrackedResourceModels;
+            foreach (var resource in trackedResources)
+            {
+                var modelHierarchy = ValidationUtilities.EnumerateModelHierarchy(resource, definitions);
+                var modelsWithLocationProp = modelHierarchy.Where(model => (definitions[model].Required?.Contains("location") == true) &&
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/LocationMustHaveXmsMutabilityExt.cs]: Yes, it should be applied to "Tracked Resources" only, I am traversing the model hierarchy for every tracked resource accordingly.

**Comment by veronicagg** [comment]
@dsgouda changes look good. I'd like to stage with rule with the others that we're adding though, so we can get a heads up to consumers about the new rules, let's figure that out before merging it.
Thanks for sending the PR for documentation to https://github.com/Azure/azure-rest-api-specs/pull/1258 (we should be moving those docs to this repo soon)

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by dsgouda** [comment]
@azuresdkci Test this please

**Comment by veronicagg** [comment]
@dsgouda changes look good to me, but I'd like to understand how we're going to be releasing new rules, I'll discuss with Samer tomorrow.

**Comment by dsgouda** [comment]
@veronicagg Given we now have a stable AutoRest build, I believe we can merge this, do approve the PR when you get time.

**Pull request merged**